tests/custom_signal_handler.rs

```diff
@@ -121,8 +121,7 @@ mod tests {
                 .unwrap_err()
                 .downcast::<Trap>()?;
             assert!(
-                trap.message()
-                    .starts_with("wasm trap: out of bounds memory access"),
+                trap.message().starts_with("wasm trap: out of bounds"),
```

Comment:
This suggests you're generating an `OutOfBounds` trap code rather than a `HeapOutOfBounds` trap code. Is that a temporary limitation, or something more fundamental? If it's temporary, could you add a TODO comment here or so?

Reply:
Hmm, I need to look into this one more. I remember making the edit thinking it was just a message change that a merge had missed, but I think you're correct that we're generating a different trap code now. We have some other failing tests on arm64 (i.e. only when running `cargo test` on arm64 hardware), so I'll revert and this just joins the club to be resolved later.